javascript/components/ShapeSource.js

```diff
@@ -80,6 +87,12 @@ class ShapeSource extends React.Component {
      * you can specify an array of string names with assets as the key `{ assets: ['pin'] }`.
     */
     images: PropTypes.object,
+
+    /**
+     * Source press listener, gets called when a user presses one of the children layers only
+     * if that layer has a higher z-index than another source layers
```

Comment:
How exactly do I specify a higher `z-index` on the `ShapeSource`?

Reply:
It's based on your layers, let's say we have two sources that share the same feature collection of points.

```javascript
<ShapeSource id='test1' source={featureCollection} onPress={...}>
    <CircleLayer id='testLayer1' style={...} />
</ShapeSource>

<ShapeSource id='test2' source={featureCollection} onPress={...}>
   <SymbolLayer id='testLayer2' aboveLayerID='testLayer1' style={...} />
</ShapeSource>
```

the onPress for `test2` would be called, since it's layers has the highest z index order. 

Internally in Mapbox GL Native, there is an array called `layers` that is kept ordered by zIndex, so the 0 index is the backgroud and N-1 is the highest zIndex layer. When you call `queryRenderedFeaturesAtPoint` or `queryRenderedFeaturesInRect` that returns all of the layers in zIndex order N-1 to 0.